src/Module/task.js

```diff
@@ -0,0 +1,133 @@
+class Task {
+  constructor() {
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+    this.taskTitle = document.querySelector('#task-description');
+    this.taskStatus = document.querySelector('#author');
+    this.taskDisplay = document.querySelector('#display');
+    this.buttonAdd = document.querySelector('#add');
+    this.clearall = document.querySelector('#clearall');
+    this.checkboxStatus = document.querySelector('#checkboxid');
+    this.render();
+  }
+
+  addTask(id, title, status) {
+    this.tasks.push({ id, title, status });
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+    this.render();
+  }
+
+  removeTask(index) {
+    this.tasks.splice(index, 1);
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+    this.render();
+  }
+
+    newEditTask(index) {  // eslint-disable-line
+    // eslint-disable-next-line
+      const toBeEdited = event.target.parentElement.parentElement.firstChild.nextElementSibling.lastChild.previousElementSibling;
+    toBeEdited.contentEditable = 'true';
+    toBeEdited.style.border = '1px solid black';
+
+      if (event.target.classList.contains('fa-solid')) {// eslint-disable-line
+
+        event.target.parentElement.parentElement.style.backgroundColor = ' #f6ee78';// eslint-disable-line
+    }
```

Comment:
- [ ] when editing a task or reloading a page the status of all the tasks is reset, please fix this before requesting another review.